File: tests/SinFun.m
classdef SinFun < Function1D
  %
  % The class DOES NOT implement eval_D and eval_DD
  % so that it uses the inherited eval_D and eval_DD
  % which compute derivative by finite difference
  %
  methods

    function self = SinFun()
    end

    function y = eval( self, x )
      y = sin(x);
    end

    function Dy = eval_D( self, x )
      disp('Use FD first derivative for SinFun');
      Dy = self.FD_eval_D(x);
    end

    function DDy = eval_DD( self, x )
      disp('Use FD second derivative for SinFun');
      DDy = self.FD_eval_DD(x);
    end

  end
end

File: _old_2019_course/julia/tests/SinFun.jl

#import Function1D
include("../libj/Function1D.jl")
function SinFun()
    return x -> sin(x)
end

#y(fun,1.57)

#Dy(fun,1.57)

#DDy(fun,1.57)
@assert y(SinFun(),pi/2) == 1
@assert y(SinFun(),0) == 0
@assert Dy(SinFun(),pi/2) == 0
@assert DDy(SinFun(),0) == 0

#print(y(fun, [0:0.2:3]))
